frontend/src/components/pages/CampOverview/EditCamperModal/EditCamperFormResponseSection.tsx

```diff
@@ -0,0 +1,175 @@
+    answer: string | undefined,
+    options: string[],
+    questionId: string,
+    isRequired: boolean,
+  ) => {
+    return (
+      <FormControl key={questionId} isRequired={isRequired}>
+        <FormLabel>
+          <Text fontSize="md" as="b">
+            {question}
+          </Text>
+        </FormLabel>
+        <RadioGroup defaultValue={answer} pb="18px">
+          <Stack direction="column">
+            {options.map((option) => (
+              <Radio
+                key={`${questionId}_${option}`}
+                value={option}
+                onChange={() => {
+                  updateFormResponse(question, option);
+                }}
+              >
+                {option}
+              </Radio>
+            ))}
+          </Stack>
+        </RadioGroup>
+      </FormControl>
+    );
+  };
+
+  const renderText = (
+    question: string,
+    answer: string,
+    questionId: string,
+    isRequired: boolean,
+  ) => {
+    return (
+      <FormControl key={questionId} minW="100%" isRequired={isRequired}>
+        <FormLabel>
+          <Text fontSize="md" as="b">
+            {question}
+          </Text>
+        </FormLabel>
+        <Textarea
+          resize="none"
+          minH="120px"
+          textAlign="start"
+          defaultValue={answer}
+          onChange={(event) => {
+            updateFormResponse(question, event.target.value);
+          }}
+          pb="18px"
+        />
+      </FormControl>
+    );
+  };
+
+  const renderMultiSelect = (
+    question: string,
+    answer: string | undefined,
+    options: string[],
+    questionId: string,
+    isRequired: boolean,
+  ) => {
+    const answeredOptions = answer?.split(",");
+
+    return (
+      <FormControl key={questionId} isRequired={isRequired} pb="18px">
+        <FormLabel>
+          <Text fontSize="md" as="b">
+            {question}
+          </Text>
+        </FormLabel>
+        <CheckboxGroup
+          defaultValue={answeredOptions}
+          onChange={(newAnswers) => {
+            const newAnswer = newAnswers.join(",");
+            updateFormResponse(question, newAnswer);
+          }}
+        >
+          <Stack direction="column">
+            {options.map((option) => (
+              <Checkbox key={`${questionId}_${option}`} value={option}>
+                {option}
+              </Checkbox>
+            ))}
+          </Stack>
+        </CheckboxGroup>
+      </FormControl>
+    );
+  };
+
+  return (
+    <VStack align="start">
+      {formQuestions.map((formQuestion) => {
+        const { question, type } = formQuestion;
+        const camperAnswerIndex = Object.keys(camperResponses).indexOf(
+          question,
+        );
+        const camperAnswer =
+          camperAnswerIndex !== -1
+            ? Object.values(camperResponses)[camperAnswerIndex]
+            : "";
+
+        switch (type) {
+          case "MultipleChoice":
+            return formQuestion.options
+              ? renderMCQ(
```

Comment:
nit: is ts smart enough that it understands if you put a null check before the switch case (in which it returns null_, and so then you don't have to null check each case?

Reply:
The null check is not in each case, so we would have to break it up for the ones that need options and the TEXT type which doesn't. I can replace the ternary with `&&` to make it a bit cleaner though